src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,309 @@
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains("<field name=\"VAR\">global:" + variableName + "</field>"))) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchLocalizationKeyUsages(Workspace workspace, String localizationKey) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().forEach(me -> {
+			GeneratableElement ge = me.getGeneratableElement();
+			if (ge != null && workspace.getGenerator().getElementLocalizationKeys(ge).contains(localizationKey)) {
+				elements.add(me);
+			} else if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains(">" + localizationKey + "</field>"))) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	/**
+	 * Scans the entire passed workspace and collects all mod elements considered to use certain value(s).
+	 *
+	 * @param workspace The project to check mod elements from for usages.
+	 * @param clazz     The class of values to be checked.
+	 * @param validIf   The predicate used to check if a field/method is considered valid.
+	 * @param condition The predicate defining the condition that the acquired values should pass.
+	 * @param <T>       The type of values to be checked.
+	 * @return List of mod elements contained in the provided workspace and considered to use certain value(s).
+	 */
+	@SuppressWarnings("unused") public static <T> List<ModElement> searchUsages(Workspace workspace, Class<T> clazz,
+			Predicate<AccessibleObject> validIf, BiPredicate<AccessibleObject, T> condition) {
+		return workspace.getModElements().stream()
+				.filter(me -> anyValueMatches(me.getGeneratableElement(), clazz, validIf, condition)).toList();
+	}
+
+	/**
+	 * Checks if values acquired from any valid fields or methods meet the specified condition.
+	 *
+	 * @param source    The object to extract values from.
+	 * @param clazz     The class of values to be checked.
+	 * @param validIf   The predicate used to check if a field/method is considered valid.
+	 * @param condition The predicate defining the condition that the acquired values should pass.
+	 * @param <T>       The type of values to be checked.
+	 * @return Whether any value extracted from valid fields/methods on the {@code source} object pass the condition.
+	 */
+	public static <T> boolean anyValueMatches(@Nullable Object source, Class<T> clazz,
+			Predicate<AccessibleObject> validIf, BiPredicate<AccessibleObject, T> condition) {
+		if (source == null)
+			return false;
+
+		for (Field field : source.getClass().getFields()) {
+			if (!Modifier.isStatic(field.getModifiers()) && (clazz.isAssignableFrom(field.getType())
+					|| validIf != null && validIf.test(field))) {

```

Comment:
I really prefer explicit parenthesis for || and &&, it makes it much easier to understand for people not knowing priority rules by heart